ocs_ci/ocs/node.py

```diff
@@ -2604,3 +2604,48 @@ def consumers_verification_steps_after_provider_node_replacement():
 
     log.info("All the consumers verification steps finished successfully")
     return True
+
+
+def generate_nodes_for_provider_worker_node_tests():
+    """
+    Generate worker node objects for the provider worker node tests.
+
+    The function generates the node objects as follows:
+    - If we have 3 worker nodes in the cluster, it generates all the 3 worker nodes
+    - If we have more than 3 worker nodes, it generates one mgr node, 2 mon nodes, and 2 osd nodes.
+    The nodes can overlap, but at least 3 nodes should be selected(and no more than 5 nodes).
+
+    Returns:
+        list: The list of the node objects for the provider node tests
+
+    """
+    wnodes = get_worker_nodes()
+    if len(wnodes) <= 3:
+        log.info(f"Generated nodes for provider node tests: {wnodes}")
+        return get_node_objs(wnodes)
+
+    mgr_node_name = pod.get_pod_node(pod.get_mgr_pods()[0]).name
+    mon_node_names = get_mon_running_nodes()
+    if mgr_node_name in mon_node_names:
+        mon_node_names.remove(mgr_node_name)
```

Comment:
though it is not default but it is possible scenario that we have 2 mgr pods now. In such case we will still have ceph_node_set with same mgr_node_name and one of mon node will be in mon_node_names

Reply:
Fixed.